pkgs/test/stdenv/default.nix

```diff
@@ -273,4 +273,45 @@ in
       };
 
   };
+
+  # This test *must* be run prior to releasing any build of either stdenv or the
+  # gcc that it exports!  This check should also be part of CI for any PR that
+  # causes a rebuild of `stdenv.cc`.
+  #
+  # When we used gcc's internal bootstrap it did this check as part of (and
+  # serially with) the gcc derivation.  Now that we bootstrap externally this
+  # check can be done in parallel with any/all of stdenv's referrers.  But we
+  # must remember to do the check.
+  #
+  gcc-stageCompare =
+    assert stdenv.cc.isGNU;
+    with pkgs;
+    # rebuild gcc using the "final" stdenv
+    let gcc-stageCompare = (gcc-unwrapped.override {
+          reproducibleBuild = true;
+          profiledCompiler = false;
+          stdenv = overrideCC stdenv (wrapCCWith {
+            cc = stdenv.cc;
+          });
+        }).overrideAttrs(_: {
+          NIX_OUTPATH_USED_AS_RANDOM_SEED = stdenv.cc.cc.out;
+        });
+    in stdenv.mkDerivation {
+      name = "gcc-stageCompare";
+      inherit (gcc-unwrapped) version;
+      dontUnpack = true;
+      dontBuild = true;
+      dontFixup = true;
+      doCheck = true;
+      installPhase = ''
+        runHook preInstall
+        touch $out
+        runHook postInstall
+      '';
+      checkPhase = ''
+        runHook preCheck
+        diff -sr ${pkgs.gcc-unwrapped.checksum}/checksums ${gcc-stageCompare.checksum}/checksums
+        runHook postCheck
+      '';
```

Comment:
I think we want to use runCommand instead

Reply:
ce47f60eda1788e63316e53ca5fa6207214a0a35